rs/ledger_suite/icp/ledger/src/main.rs

```diff
@@ -804,6 +803,14 @@ fn post_upgrade(args: Option<LedgerCanisterPayload>) {
         });
     }
 
+    if ledger.ledger_version > LEDGER_VERSION {
```

Comment:
Should it be allowed to skip a version? If not (I assume so), then this check could be further hardened.

Reply:
The idea behind this check is that previous version cannot know about changes made by future version and decide if it allows downgrade or not. Future version can do that, e.g. V2 allows migration from V1, V0 and earlier, V4 might allow migration from V2 if allowance migration code is left in place. V3 does not allow migration from V1 by failing to read the raw stable memory. If future version cannot determine if skipping a version is allowed by reading the ledger state (as in the V1->V3 example), we can then extend this check for that version.